File: src/features/typing/data/java/longest_consecutive_sequence.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }
        int best = 0;
        for (int num : set) {
            if (set.contains(num - 1)) {
                continue;
            }
            int length = 1;
            while (set.contains(num + length)) {
                length++;
            }
            best = Math.max(best, length);
        }
        return best;
    }
}

File: src/features/typing/data/dart/longest_consecutive_sequence.dart
class Solution {
  int longestConsecutive(List<int> nums) {
    final set = nums.toSet();
    var best = 0;
    for (final num in set) {
      if (set.contains(num - 1)) {
        continue;
      }
      var length = 1;
      while (set.contains(num + length)) {
        length++;
      }
      if (length > best) {
        best = length;
      }
    }
    return best;
  }
}
